src/product_representations.jl

```diff
@@ -310,7 +310,7 @@ function allocate(
     return ProductArray(ShapeSpec, allocate(x.data, T), x.reshapers)
 end
 allocate(p::ProductArray, ::Type{T}, s::Size{S}) where {S,T} = Vector{T}(undef, S)
-allocate(p::ProductArray, ::Type{T}, s::Integer) where {T} = Vector{T}(undef, s)
+allocate(p::ProductArray, ::Type{T}, s::Int64) where {T} = Vector{T}(undef, s)
```

Comment:
Is it necessary to be that specific? What was the problem with `Integer`?

Reply:
There was an ambiguity because `allocate` has `Int64` in ManifoldsBase in one place, I'll fix this later. Right now I'm transitioning decorators in ManifoldsBase to the new parent/transparent/intransparent dispatch based on different functions.